colibre/registration.py

```diff
@@ -24,333 +24,458 @@
-    setattr(self, f"sfr_halo_mass_{aperture_size}_kpc", sfr_M200)
+        # Name (label) of the derived field
+        ssfr.name = f"Specific SFR ({aperture_size} kpc)"
 
-# Now metallicities relative to different units
-solar_metal_mass_fraction = 0.0126
-twelve_plus_log_OH_solar = 8.69
-minimal_twelve_plus_log_OH = 7.5
+        # Mask for the passive objects
+        is_passive = unyt.unyt_array(
+            (ssfr < 1.01 * marginal_ssfr).astype(float), units="dimensionless"
+        )
+        is_passive.name = "Passive Fraction"
 
-for aperture_size in aperture_sizes:
-    try:
-        metal_mass_fraction_star = (
-            getattr(catalogue.apertures, f"zmet_star_{aperture_size}_kpc")
-            / solar_metal_mass_fraction
+        # Mask for the active objects
+        is_active = unyt.unyt_array(
+            (ssfr > 1.01 * marginal_ssfr).astype(float), units="dimensionless"
         )
-        metal_mass_fraction_star.name = f"Star Metallicity $Z_*$ rel. to $Z_\\odot={solar_metal_mass_fraction}$ ({aperture_size} kpc)"
-        setattr(
-            self,
-            f"star_metallicity_in_solar_{aperture_size}_kpc",
-            metal_mass_fraction_star,
+        is_active.name = "Active Fraction"
+
+        # Get the specific star formation rate (per halo mass instead of stellar mass)
+        sfr_M200 = star_formation_rate / halo_mass
+        sfr_M200.name = "Star formation rate divided by halo mass"
+
+        # Register derived fields with specific star formation rates
+        setattr(self, f"specific_sfr_gas_{aperture_size}_kpc", ssfr)
+        setattr(self, f"is_passive_{aperture_size}_kpc", is_passive)
+        setattr(self, f"is_active_{aperture_size}_kpc", is_active)
+        setattr(self, f"sfr_halo_mass_{aperture_size}_kpc", sfr_M200)
+
+    return
+
+
+def register_star_metallicities(self, catalogue):
+
+    # Define aperture size in kpc
+    aperture_sizes = [30, 100]
+
+    # Solar metallicity
+    solar_metal_mass_fraction = 0.0126
+
+    # Loop over apertures
+    for aperture_size in aperture_sizes:
+
+        try:
+            metal_mass_fraction_star = (
+                getattr(catalogue.apertures, f"zmet_star_{aperture_size}_kpc")
+                / solar_metal_mass_fraction
+            )
+            metal_mass_fraction_star.name = (
+                f"Star Metallicity $Z_*$ rel. to "
+                f"$Z_\\odot={solar_metal_mass_fraction}$ ({aperture_size} kpc)"
+            )
+            setattr(
+                self,
+                f"star_metallicity_in_solar_{aperture_size}_kpc",
+                metal_mass_fraction_star,
+            )
+        except AttributeError:
+            pass
+
+    return
+
+
+def register_gas_phase_metallicities(self, catalogue):
+
+    # Define aperture size in kpc
+    aperture_sizes = [30, 100]
+
+    # Metallicities relative to different units
+    solar_metal_mass_fraction = 0.0126
+    twelve_plus_log_OH_solar = 8.69
+    minimal_twelve_plus_log_OH = 7.5
```

Comment:
Parameters like this that have fixed values should be either globals or passed as arugments to the function.

Reply:
Now these are globals